sdk/translation/Azure.AI.Translation.Document/src/DocumentTranslationModelFactory.cs

```diff
@@ -0,0 +1,160 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Azure.Core;
+using Azure.AI.Translation.Document.Models;
+namespace Azure.AI.Translation.Document
+{
+    /// <summary>
+    /// Model factory that enables mocking for the Document Translation Library.
+    /// </summary>
+    [CodeGenModel("BatchDocumentTranslationModelFactory")]
+    public static partial class DocumentTranslationModelFactory
+    {
+        #region Errors
+        /// <summary>
+        /// Initializes a new instance of <see cref="Document.DocumentTranslationError"/> for mocking purposes.
+        /// </summary>
+        /// <param name="errorCode">Sets the <see cref="DocumentTranslationError.ErrorCode"/> property.</param>
+        /// <param name="message">Sets the <see cref="DocumentTranslationError.Message"/> property.</param>
+        /// <param name="target">Sets the <see cref="DocumentTranslationError.Target"/> property.</param>
+        /// <returns>A new instance of <see cref="Document.DocumentTranslationError"/> for mocking purposes.</returns>
+        public static DocumentTranslationError DocumentTranslationError(
+            DocumentTranslationErrorCode  errorCode,
+            string message,
+            string target
+            )
+        {
+            return new DocumentTranslationError(errorCode, message, target, default);
+        }
+        #endregion Errors
+
+        #region Statuses
+        /// <summary>
+        /// Initializes a new instance of <see cref="Document.DocumentStatus"/> for mocking purposes.
+        /// </summary>
+        /// <param name="id">Sets the <see cref="DocumentStatus.Id"/> property.</param>
+        /// <param name="translatedDocumentUri">Sets the <see cref="DocumentStatus.TranslatedDocumentUri"/> property.</param>
+        /// <param name="sourceDocumentUri">Sets the <see cref="DocumentStatus.SourceDocumentUri"/> property.</param>
+        /// <param name="createdOn">Sets the <see cref="DocumentStatus.CreatedOn"/> property.</param>
+        /// <param name="lastModified">Sets the <see cref="DocumentStatus.LastModified"/> property.</param>
+        /// <param name="status">Sets the <see cref="DocumentStatus.Status"/> property.</param>
+        /// <param name="translatedTo">Sets the <see cref="DocumentStatus.TranslatedTo"/> property.</param>
+        /// <param name="progress">Sets the <see cref="DocumentStatus.Progress"/> property.</param>
+        /// <param name="charactersCharged">Sets the <see cref="DocumentStatus.CharactersCharged"/> property.</param>
+        /// <returns>A new instance of <see cref="Document.DocumentStatus"/> for mocking purposes.</returns>
+        public static DocumentStatus DocumentStatus(
+            string id,
+            Uri translatedDocumentUri,
```

Comment:
I am nitpicking here. Could we swap the `translatedDocumentUri` and `sourceDocumentUri` so the first one is the `sourceDocumentUri`? I think it will make more sense to first see source and then target -> translatedDocumentUri